src/main/java/cholog/wiseshop/api/campaign/service/CampaignService.java

```diff
@@ -127,4 +131,25 @@ public List<MemberCampaignResponse> readMemberCampaign(Member member) {
                 return MemberCampaignResponse.of(campaign, products.getFirst());
             }).toList();
     }
+
+    public List<ReadCampaignResponse> searchByProductName(String keyword, int page, int size) {
+        keyword = keyword.trim();
+        Pageable pageable = PageRequest.of(page, size, Sort.by("id").ascending());
+        Page<Product> pages = productRepository.findByNameContaining(keyword, pageable);
+        return pages.getContent().stream()
+            .map(product -> {
+                Campaign campaign = product.getCampaign();
+                return new ReadCampaignResponse(
+                    campaign.getId(),
+                    campaign.getStartDate().toString(),
+                    campaign.getEndDate().toString(),
+                    campaign.getGoalQuantity(),
+                    campaign.getSoldQuantity(),
+                    product.getStock().getTotalQuantity(),
+                    campaign.getState(),
+                    new ProductResponse(product),
+                    campaign.getMember().orElse(Member.createEmpty()).getId()
+                );
+            }).toList();

```

Comment:
```suggestion
        return pages.getContent().stream()
            .map(product -> {
                Campaign campaign = product.getCampaign();
                Member member = campaign.getMember().orElse(Member.createEmpty());
                return new ReadCampaignResponse(
                    campaign.getId(),
                    campaign.getStartDate().toString(),
                    campaign.getEndDate().toString(),
                    campaign.getGoalQuantity(),
                    campaign.getSoldQuantity(),
                    product.getStock().getTotalQuantity(),
                    campaign.getState(),
                    new ProductResponse(product),
                    member.getId()
                );
            }).toList();

```
멤버는 바깥에서 분리하는건 어떤가용??